src/stdlib_bitset_64.fypp

```diff
@@ -0,0 +1,1145 @@
+            return
+        else
+            new % num_bits = bits
+            do i=0, bits-1
+                k = start_pos + i
+                if ( btest( old % block, k ) ) &
+                    new % block = ibset(new % block, i)
+            end do
+        end if
+
+        if ( present(status) ) status = success
+
+    end subroutine extract_64
+
+
+    elemental module subroutine flip_bit_64(self, pos)
+!
+!     Flips the value at the POS position in SELF, provided the position is
+!     valid. If POS is less than 0 or greater than BITS(SELF)-1, no value is
+!     changed.
+!
+        class(bitset_64), intent(inout) :: self
+        integer(bits_kind), intent(in)  :: pos
+
+        if ( pos < 0 .OR. pos > self % num_bits-1 ) return
+
+        if ( btest( self % block, pos ) ) then
+            self % block = ibclr( self % block, pos )
+        else
+            self % block = ibset( self % block, pos )
+        end if
+
+    end subroutine flip_bit_64
+
+
+    pure module subroutine flip_range_64(self, start_pos, stop_pos)
+!
+!     Flips all valid bits from the START_POS to the STOP_POS positions in
+!     SELF. If STOP_POS < START_POS no bits are flipped. Positions less than
+!     0 or greater than BITS(SELF)-1 are ignored.
+!
+        class(bitset_64), intent(inout) :: self
+        integer(bits_kind), intent(in)  :: start_pos, stop_pos
+
+        integer(bits_kind) :: end_bit, start_bit
+
+        start_bit = max( 0, start_pos )
+        end_bit   = min( stop_pos , self % num_bits-1 )
+        call mvbits( not(self % block),       &
+                     start_bit,               &
+                     end_bit - start_bit + 1, &
+                     self % block,            &
+                     start_bit )
+
+    end subroutine flip_range_64
+
+
+    module subroutine from_string_64(self, string, status)
+!     Initializes the bitset `self` treating `string` as a binary literal
+!     `status` may have the values:
+!         `success` - if no problems were found,
+!         `alloc_fault` - if allocation of the bitset failed
+!         `char_string_too_large_error` - if `string` was too large, or
+!         `char_string_invalid_error` - if string had an invalid character.
+        class(bitset_64), intent(out)  :: self
+        character(*), intent(in)       :: string
+        integer, intent(out), optional :: status
+
+        character(*), parameter :: procedure = 'FROM_STRING'
+        integer(bits_kind) :: bit
+        integer(int64)     :: bits
+        character(1)       :: char
+
+        bits = len(string, kind=int64)
+        if ( bits > 64 ) then
+            call error_handler( 'STRING was too long for a ' //      &
+                                'BITSET_64 SELF.',                   &
+                                char_string_too_large_error, status, &
+                                module_name, procedure )
+            return
+        end if
+        self % num_bits = bits
+        do bit = 1, bits
```

Comment:
I get this warning compiling it:
```
/home/ian/stdlib/src/stdlib_bitset_64.f90:437:19:

  437 |         do bit = 1, bits
      |                   1
Warning: Possible change of value in conversion from INTEGER(8) to INTEGER(4) at (1) [-Wconversion]
```
Maybe changing to this:
```suggestion
        do bit = 1_bits_kind, bits
```
Or modifying the kind of the variable `bit` to other than `bits_kind` since it's a simple iterator

Reply:
If you check on the use of bits, it is declared int64 to always ensure that an int32 for bit is adequate. The warning is not for the 1 in `do bit = 1, bits`, but rather for the conversion of `bits` to `bit`. The following tests ensure that int32 is adequate.

        bits = len(string, kind=int64)
        if ( bits > 64 ) then
 
in stdlib_bitset_64.fypp or

        bits = len(string, kind=int64)
        if ( bits > huge(0_bits_kind) ) then

in stdlib_bitset_large.fypp. So the warning is spurious. I can, if you want, shut off the warning by making bit an int64.

Reply:
That's correct, it's `bit` vs `bits` conversion that raises this warning. Is there any reason to `bit` be of kind `bits_kind`, in this function? If not, I think it should be a `int64` as well.